tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -174,4 +176,297 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+		e2eVSphere.verifyVolumeCompliance(volumeID, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+
+		defer func() {
+			ginkgo.By("Delete the PVCs created in step 3")
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volumeID)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		}()
+		// TODO: Verify no orphan volumes are created
+	})
+
 })
+
+//fillVolumesInPods fills the volumes in pods upto given percentage
+func fillVolumeInPods(f *framework.Framework, pods []*v1.Pod) {
+	for _, pod := range pods {
+		size, err := getFSSizeMb(f, pod)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		writeRandomDataOnPod(f, pod, size-66) // 66 is just from my observation and not technically derived
+	}
+}
+
+//writeRandomDataOnPod runs dd on the given pod and write count in Mib
+func writeRandomDataOnPod(f *framework.Framework, pod *v1.Pod, count int64) {
+	cmd := []string{"--namespace=" + pod.Namespace, "-c", pod.Spec.Containers[0].Name, "exec", pod.Name, "--",
+		"/bin/sh", "-c", "dd if=/dev/urandom of=/mnt/volume1/f1 bs=1M count=" + strconv.FormatInt(count, 10)}
+	_ = framework.RunKubectlOrDie(pod.Namespace, cmd...)
+}
+
+//setVpxdTaskTimeout sets vpxd task timeout to given number of seconds
+func setVpxdTaskTimeout(taskTimeout int) {
+	vcAddress := e2eVSphere.Config.Global.VCenterHostname + ":" + sshdPort
+	timeoutMatches := false
+	diffTimeoutExists := false
+
+	grepCmd := "grep '<timeout>' /etc/vmware-vpx/vpxd.cfg"
+	framework.Logf("Invoking command '%v' on vCenter host %v", grepCmd, vcAddress)
+	result, err := fssh.SSH(grepCmd, vcAddress, framework.TestContext.Provider)
+	if err != nil {
+		fssh.LogResult(result)
+		err = fmt.Errorf("couldn't execute command: %s on vCenter host %v: %v", grepCmd, vcAddress, err)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+
+	sshCmd := fmt.Sprintf(
+		"sed -i 's/<task>/<task>\\n    <timeout>%v<\\/timeout>/' /etc/vmware-vpx/vpxd.cfg", taskTimeout)
+	if result.Code == 0 {
+		grepCmd2 := fmt.Sprintf("grep '<timeout>%v</timeout>' /etc/vmware-vpx/vpxd.cfg", taskTimeout)
+		framework.Logf("Invoking command '%v' on vCenter host %v", grepCmd2, vcAddress)
+		result2, err := fssh.SSH(grepCmd2, vcAddress, framework.TestContext.Provider)
+		if err != nil {
+			fssh.LogResult(result)
+			err = fmt.Errorf("couldn't execute command: %s on vCenter host %v: %v", grepCmd2, vcAddress, err)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+		if result2.Code == 0 {
+			timeoutMatches = true
+		} else {
+			diffTimeoutExists = true
+		}
+	} else {
+		if taskTimeout == 0 {
+			timeoutMatches = true
+		}
+	}
+	if timeoutMatches {
+		framework.Logf("vpxd timeout already matches, nothing to do ...")
+		return
+	}
+	if diffTimeoutExists {
```

Comment:
Can't we have only one bool variable timeoutMatches here? If vpxd timout is same as we want, then we can set it as true otherwise set it as false and execute this piece of block.

Reply:
I want to handle the following cases,
1. timeout is not set, we want to set it
2. timeout is set we want to clear it
3. a different timeout is set, we want to change it
hence using two variables